ui/desktop/electron-app/src/cli/path.js

```diff
@@ -8,18 +8,21 @@ const { isWindows } = require('../helpers/platform.js');
 const isDev = require('electron-is-dev');
 const { existsSync } = require('node:fs');
 
+const binaryName = isWindows() ? 'boundary.exe' : 'boundary';
+const builtInCliPath = isDev
+  ? path.resolve(__dirname, '..', '..', 'cli', binaryName)
+  : path.resolve(process.resourcesPath, 'cli', binaryName);
+
+// Return true if the CLI in usage is the built in. False if relies on system CLI.
+const isBuiltInCli = existsSync(builtInCliPath);
+
 module.exports = {
   /**
    * Returns Boundary CLI path if the CLI is built in or the Boundary binary name
    * if not, so we assume boundary is available within user $PATH.
    */
   path: () => {
-    const binaryName = isWindows() ? 'boundary.exe' : 'boundary';
-    const builtInCliPath = isDev
-      ? path.resolve(__dirname, '..', '..', 'cli', binaryName)
-      : path.resolve(process.resourcesPath, 'cli', binaryName);
-    const isBuiltInCli = existsSync(builtInCliPath);
-
     return isBuiltInCli ? builtInCliPath : binaryName;
   },
+  isBuiltInCli,
```

Comment:
This will only be called once I believe when this file is `require`d which I kind of like since it doesn't need to keep doing the check each time this is called but is there any concern we would ever need to re-calculate this during runtime? I'm thinking no but just wanted to make sure we're doing this intentionally. 

That also brings up should `path()` still be a function as well

Reply:
@ZedLi 

That is correct, this is intentionally. When working on this, I realize path() was call (therefore re-calculate) several ties just to execute DC. I found that unnecessary and wanted isBuiltInCli to not follow that approach.

I don't find any concern following this approach when it comes to CLI checks, let me explain my thinking here, perhaps I missed something.

- In case DC includes CLI: re-check this is unncessary, since the CLI will always live in the same place. If any day we want to change CLI localtion, we will release a new DC, so no problem here.
- In case DC does not include CLI, we rely on user's path: all users will use the instaler, so it is safe to assume most users will not be changing CLI location manually.
If an advanced user decides to change CLI location, I assume, it will manually update boundary location within user path and "restart" boundary, boundary client agent and DC client.

That is correct, as per the thinking explained above, I do strongly believe path() should be just a const and not a function.
I wanted to wait on the windows bug resolution before introducing such change, but after seeing Dharini's work, we will need to resolve conflicts anyway, so adding it within this PR.
The path change, has been successfully tested manually and using the DC e2e tests.

Let me know if you think otherwise, happy to discuss other approaches.